vm/channel.go

```diff
@@ -63,9 +63,22 @@ import (
 // ```
 type ChannelObject struct {
 	*baseObj
-	Chan chan int
+	Chan         chan int
+	ChannelState int
 }
 
+// Channel's state.
+// To Goby language contributors: Golang's channels should be carefully handled because:
+// - You cannot write to closed channels, or got a panic.
+// - You cannot close closed channels, or got a panic.
+// - You cannot write to nil channels, or causes a deadlock.
+// - You cannot read nil channels, or causes a deadlock.
+// Ref: https://beatsync.net/main/log20150325.html
+const (
+	OPEN = iota
```

Comment:
Probably want to prefix with some identifier that these refer to channel states. Also, prefer the go convention of using camel case. These also probably shouldn't be exported values. Packages outside of vm should not be touching these state values.

Reply:
Thank you! I missed the point.

Reply:
Sorry for the delay: updated the constants